### DISCUSSION TITLE
prefix and osm2pgsql_properties for multiple different imports

### DISCUSSION THREAD

**Discussion by danielengels** [osm2pgsql-dev/osm2pgsql#2280] [Q&A] [answered]
Currently I run an older version of osm2qsql and want to update to the latest version.
I already get the new flexoutput with the compatible.lua running but I get in trouble with the prefix and the new osm2pgsql_properties.

1. I have to set the prefix 2 times. 1st time in the lua script and 2nd to the commandline. Is that correct? Do I don't get something. But here i can make a work around and create the lua script automatically and change the prefix.

2. At the moment I have many quiet small extraction of some areas in my database. The import of these areas are always in the same database with different prefixes. When I need an update of the imports i always import the whole data again. But here I get in trouble with osm2pgsql_properties, as here the name don't seems to be changeable. As the import will start from a webfronend that a few users are using, the import could be also happen in the same time. This causes errors. The only workarround for this seems for me at the moment creating separate schemas or databases. But is there a better way of doing?

**Comment by joto** [comment, marked as answer]
1. The prefix for the middle tables can only be set on the command line. If you don't use `--slim` you don't have them and don't need to set the prefix on the command line. There is no magic prefix any more for flex tables. If you want all tables name to have the same prefix, you can do that in your Lua script, that's what the `compatible.lua` does to make it easier to switch from the old way of doing things. Lua is very flexible in what it can do, you can read [environment variables](https://osm2pgsql.org/doc/manual.html#accessing-environment-variables-from-lua), for instance. For automated environments you can use this to set the prefix (or anything else) from outside the Lua script.
2. No, there is no better way to do this. You'll have to use different schemas or databases.

**Reply by danielengels** [reply]
Thank you for your fast answer!
I guess I need the --slim mode else I had trouble before with some imports.
local prefix = os.getenv("prefix") in the lua and in the commandline -p $prefix shouldn't be a problem I guess. 
The different schemas or databases will make me more problems, but I guess that is all still easy solvable.
Now I will try to find out, how to lua is working exactly and if I will be able to merge some data with strange associatedStreet relations ;)
